src/index.ts

```diff
@@ -137,6 +148,32 @@ export class WaxJS {
     return this.user.account;
   }
 
+  private async getChainInfoByChainName(chainName: string): Promise<any> {
+    const response = await this.rpc.get_table_rows({
+      json: true,
+      code: "registry.wax",
+      scope: "registry.wax",
+      table: "chains",
+    });
+    const chainInfo: any | null = response.rows.find(row => row.chain_name === chainName);
+
+    if (!chainInfo) {
+      // TODO: handle case chain info not exists
+    }
+
+    return chainInfo;
+  }
+
+  public async setChainName(chainName: string): Promise<void> {
```

Comment:
`setupChain` might be more appropriate since we are setting more than just the name

Reply:
At the top of this function, let's just check if chainName is null or not defined, and set up the chainName and chainId with default values right here:
```
if(!ChainName) {
  this.chainName = null;
  this.chainId = null;
  return;
}
```

Reply:
updated base on comment 👍